### PR TITLE
Add eslint rules for safe use of TS enums

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
The new rules require that all enum values have initializers and that those initializers are literal values. TS enums work different from Flow enums, in particular they default to numeric value, e.g. 0, 1, 2, ... instead of a string representation of the identifier.

Using numeric values is unsafe because inserting or removing enum variants will result in the numeric values of the remaining variants changing which will likely break code.

Issue: None

## Test plan:
- yarn lint

### CODE REVIEW

File: packages/perseus/src/types.ts [hunk 1]
```diff
@@ -352,7 +352,7 @@ export type APIOptionsWithDefaults = Readonly<
 export type LinterContextProps = {
     contentType: string;
     highlightLint: boolean;
-    paths: ReadonlyArray<string>;
+    path: ReadonlyArray<string>;
```
Comment (kevinbarabash) [inline packages/perseus/src/types.ts:355]: This change is not related to the rest of this PR and will be reverted.

**Comment by jeremywiebe** [comment]
Could you add to the PR description why not using initializers is unsafe (just so that when this PR lands we have the reasoning for posterity)? :)

**Review by jeremywiebe — APPROVED** [review]
